app/controllers/stories_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class StoriesController < ApplicationController
+  before_action :update_top_stories, only: :index
+
+  def index
+    @stories = Story.where(hacker_news_id: top_stories_ids).includes(:users)
+  end
+
+  private
+
+  def top_stories
+    @top_stories ||= Story.where(hacker_news_id: top_stories_ids)
+  end
+
+  def top_stories_ids
+    @top_stories_ids ||= HackerNews.new.top_stories_ids[0..50]
+  end
+
+  def missing_stories
+    top_stories_ids - top_stories.pluck(:hacker_news_id)
+  end
+
+  def update_top_stories
```

Comment:
If a new story id appears in the top news array, store it locally so it can be starred.